File: Algorithms/Greedy/LargestNumber.py

from functools import cmp_to_key

def get_largest_number(nums):

    values = list(map(str, nums))

    values.sort(key=cmp_to_key( lambda a, b: 1 if (b + a) > (a + b) else -1 ))

    return int("".join(values))

def main():

    nums = [22, 225, 53, 19, 100]

    print(get_largest_number(nums))

if __name__ == '__main__':
    main()

File: Algorithms/Greedy/LargestNumber.js
"use strict";

let getLargestNumber = (nums) => {

    let values = nums.map(String);

    values.sort((a, b) => (a + b) > (b + a) ? -1 : 1);

    return parseInt(values.join(''));

}

let nums = [22, 225, 53, 19, 100];

console.log(getLargestNumber(nums));
